baseband/gsb/tests/test_gsb.py

```diff
@@ -270,241 +331,286 @@ def test_rawdump_frame(self, tmpdir):
+            with open(SAMPLE_RAWDUMP_HEADER, 'rt') as ft, \
+                    open(SAMPLE_RAWDUMP, 'rb') as fraw:
+                frame1 = gsb.GSBFrame.fromfile(ft, fraw, payloadsize=2**12,
+                                               bps=4)
+            assert fh_r.header0.time == fh_r.start_time
+            assert fh_r.header0 == frame1.header
+            assert np.all(fh_r.read(fh_r.samples_per_frame) == frame1.data)
+            # Seek last offset.
+            with open(SAMPLE_RAWDUMP_HEADER, 'rt') as ft, \
+                    open(SAMPLE_RAWDUMP, 'rb') as fraw:
+                ft.seek(frame1.header.seek_offset(9))
+                fraw.seek(9 * fh_r._payloadsize)
+                frame10 = gsb.GSBFrame.fromfile(ft, fraw, payloadsize=2**12,
+                                                bps=4)
+            assert fh_r._last_header == frame10.header
+            fh_r.seek(-10, 2)
+            assert np.all(fh_r.read(10) == frame10.data[-10:])
+            # Check validity of current and stopping time
+            assert abs(fh_r.stop_time -
+                       Time('2015-04-27T13:15:02.516582640')) < 1.*u.ns
+            assert abs(fh_r.stop_time - fh_r.time) < 1.*u.ns
             fh_r.seek(0)
-            fh_r.read(out=out)
-            # check that non-squeezed output is identical in content
-            fh_r.squeeze = False
-            out2 = np.zeros_like(self.data)
+            data1 = fh_r.read()
+            assert fh_r.tell() == len(data1)
+            assert data1.shape[0] == fh_r.size
+            fh_r.seek(0)
+            out1 = np.empty_like(data1)
+            fh_r.read(out=out1)
+            assert np.all(out1 == data1)
+            fh_r.seek(0)
+            fh_r.squeeze = True
+            out2 = np.empty((fh_r.size,) + fh_r.sample_shape)
             fh_r.seek(0)
             fh_r.read(out=out2)
-        assert np.all(data == self.data.ravel())
-        assert np.all(out == self.data.squeeze())
-        assert np.all(out2 == self.data)
-
-        with io.open(str(tmpdir.join('test_time.dat')), 'w+b') as sh, \
-                open(str(tmpdir.join('test.dat')), 'w+b') as sp, \
-                gsb.open(sh, 'ws', raw=sp, sample_rate=4096*u.Hz,
-                         samples_per_frame=payload.nsample, **header) as fh_w:
-            fh_w.write(self.data.ravel())
-            fh_w.write(self.data.ravel())
-            assert fh_w.tell() == 2 * len(self.data)
-            assert_quantity_allclose(fh_w.tell(unit=u.s), 1. * u.s)
+            assert np.all(out2 == out1.squeeze())
+            # To compare with directly psasing samples_per_frame below.
+            spf_from_payloadsize = fh_r.samples_per_frame
+            # For testing proper file closing below.
+            header0 = fh_r.header0
+
+        with gsb.open(SAMPLE_RAWDUMP_HEADER, mode='rs', raw=SAMPLE_RAWDUMP,
+                      samples_per_frame=2**13) as fh_r, \
+                io.open(str(tmpdir.join('test_time.timestamp')), 'w+b') as sh,\
```

Comment:
no need for `io.`

Reply:
In Python 2 removing `io` gives:

```
 AttributeError: 'file' object has no attribute 'readable'
```

which I thought was why the old `test_gsb.py` also only opened timestamp files with `io`?

Reply:
If it is a python2 thing, can we remove the `io.` here, but add at the top
```
if six.PY2:
    from io import open
```
That way we can remove the clutter when we're python3-only (not too long from now...)